apps/browser/qml/pages/components/PopUpMenuItem.qml

```diff
@@ -0,0 +1,222 @@
+                iconSource: "image://theme/icon-m-tab-new"
+                //% "New tab"
+                text: qsTrId("sailfish_browser-la-new_tab")
+                onClicked: {
+                    webView.privateMode = false
+                    overlay.toolBar.enterNewTabUrl()
+                }
+            }
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-incognito-new"
+                //% "New private tab"
+                text: qsTrId("sailfish_browser-la-new_private_tab")
+                onClicked: {
+                    webView.privateMode = true
+                    overlay.toolBar.enterNewTabUrl()
+                }
+            }
+        }
+
+        Column {
+            width: parent.width
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-search-on-page"
+                enabled: webView.contentItem
+                //% "Search on page"
+                text: qsTrId("sailfish_browser-la-search_on_page")
+
+                onClicked: {
+                    overlay.toolBar.findInPageActive = true
+                    overlay.toolBar.findInPage()
+                }
+            }
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-add-to-grid"
+                //% "Add to apps grid"
+                text: qsTrId("sailfish_browser-la-add_to_apps_grid")
+                onClicked: {
+                    overlay.animator.showChrome()
+                    pageStack.animatorPush("AddToAppGridDialog.qml",
+                                                  {
+                                                      "url": url,
+                                                      "title": title,
+                                                      "desktopBookmarkWriter": desktopBookmarkWriter,
+                                                      "bookmarkWriterParent": pageStack
+                                                  })
+                }
+            }
+
+            OverlayListItem {
+                height: Theme.itemSizeSmall
+                iconWidth: root.iconWidth
+                horizontalOffset: root.horizontalOffset
+                iconSource: "image://theme/icon-m-file-download-as-pdf"
+                //% "Save web page as PDF"
+                text: qsTrId("sailfish_browser-la-save_as-pdf")
+
+                onClicked: {
+                    overlay.animator.showChrome()
+                    if (DownloadManager.pdfPrinting) {
```

Comment:
This is working fine as it is, but I notice that `DownloadManager.pdfPrinting` is a `Q_PROPERTY` with signals when it changes. Given this, you could consider greying-out the button while printing is taking place.

Reply:
@llewelld Done. Add:

>  !DownloadManager.pdfPrinting
                opacity: enabled ? 1.0 : 0.5`